include/grpc++/impl/codegen/async_unary_call.h

```diff
@@ -201,7 +208,8 @@ class ClientAsyncResponseReader final
 /// Async server-side API for handling unary calls, where the single
 /// response message sent to the client is of type \a W.
 template <class W>
-class ServerAsyncResponseWriter final : public ServerAsyncStreamingInterface {
+class ServerAsyncResponseWriter final
+    : public internal::ServerAsyncStreamingInterface {
  public:
   explicit ServerAsyncResponseWriter(ServerContext* ctx)
```

Comment:
So the Server constructors are different from the Client constructors because the Server constructors are actually part of the application-level API while the client constructors are invoked from generated code only. A client object gets created in response to a call on a stub (lives in generated code) whereas a server object gets created in order to request (pre-register) a call at the server. So they're pretty different.